script.js

```diff
@@ -1,100 +1,332 @@
-    let cantidad = parseFloat(prompt(`Ingrese la cantidad requerida de ${nombre} por unidad de producto (ej: 150):`));
-    let unidad = prompt(`Ingrese la unidad de medida para ${nombre} (ej: litro, metro, caja):`);
-    let unidadesPorSeleccionada = parseFloat(prompt(`Ingrese cantidad de unidades estándar que hay por cada ${unidad} (ej: 12 si una caja tiene 12 unidades):`));
-
-    if (
-  !nombre || isNaN(costoUnitario) || isNaN(cantidad) ||
-  !unidad || isNaN(unidadesPorSeleccionada)) {
-      alert("Por favor, ingrese datos válidos.");
-}else{
-        calculadora.agregarMaterial({ nombre, costoUnitario, cantidad, unidad, unidadesPorSeleccionada });
-    alert(" Material agregado correctamente.");
+// se limpia y actualiza la tabla de gastos fijos con los datos actuales
+function actualizarTablaGastosFijos() {
+  tablaGastosFijos.innerHTML = "";
+  calculadora.gastosFijos.forEach((g, i) => mostrarGastoFijoEnTabla(g, i));
 }
-        seguirMaterial = confirm("¿Quiere agregar otro material?");
+
+// Evento para agregar material
+// Valida los datos, agrega el material, actualiza la tabla y guarda en localStorage
+formProducto.addEventListener("submit", function (e) {
+  e.preventDefault();
+  const nombre = document.getElementById("nombre-material").value.trim();
+  const costoUnitario = parseFloat(document.getElementById("costo-material").value);
+  const cantidad = parseFloat(document.getElementById("cantidad").value);
+  const unidad = document.getElementById("unidad-material").value;
+  const unidadesPorSeleccionada = parseFloat(document.getElementById("cantidad-equivalente").value);
+
+  if (nombre && !isNaN(costoUnitario) && !isNaN(cantidad) && unidad && !isNaN(unidadesPorSeleccionada)) {
+    const material = { nombre, costoUnitario, cantidad, unidad, unidadesPorSeleccionada };
+    calculadora.agregarMaterial(material);
+   actualizarTablaMateriales(); 
+    guardarDatosEnStorage();
+
+    //Se limpian los campos
+    formProducto.reset();
+
+     // Se muestra modal dependiendo si se completaron o no los campos
+  const modalMaterial = new bootstrap.Modal(document.getElementById('agregarMaterial'));
+  modalMaterial.show();
+} 
+  }
+);
+
+
+//Evento para editar material
+tablaMateriales.addEventListener("click", (e) => {
+  if (e.target.dataset.tipo === "editar-material") {
+    const index = parseInt(e.target.dataset.index);
+    const m = calculadora.materiales[index];
+
+   // Prellena el formulario con los datos y elimina temporalmente el material para reemplazarlo
+    document.getElementById("nombre-material").value = m.nombre;
+    document.getElementById("costo-material").value = m.costoUnitario;
+    document.getElementById("cantidad").value = m.cantidad;
+    document.getElementById("unidad-material").value = m.unidad;
+    document.getElementById("cantidad-equivalente").value = m.unidadesPorSeleccionada;
+
+    // Eliminar y esperar nuevo submit
+    calculadora.materiales.splice(index, 1);
```

Comment:
Delega la responsabilidad a calculadora:
```suggestion
calculadora.eliminarMaterial(index) 
```